### ISSUE TITLE
CI failure at CoreDNS deployment

### ISSUE THREAD

**Issue by okozachenko1203** [vexxhost/atmosphere#453]
`ceph` CI workflow is failing at the `molecule converge` step. More specifically, the ansible task to install CoreDNS is timeout.
Need to fix it.

**Comment by okozachenko1203** [comment]
https://github.com/vexxhost/atmosphere/pull/454
In this way, we can have a path for CI troubleshooting at anytime

**Comment by okozachenko1203** [comment]
CoreDNS pods are failed scheduling
```
Events:
  Type     Reason                  Age                 From               Message
  ----     ------                  ----                ----               -------
  Warning  FailedScheduling        36m                 default-scheduler  0/1 nodes are available: 1 node(s) had taint {node.kubernetes.io/not-ready: }, that the pod didn't tolerate.
  Normal   Scheduled               35m                 default-scheduler  Successfully assigned kube-system/coredns-8558b7dd95-t76zd to instance
  Warning  FailedCreatePodSandBox  34m                 kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "4ef235a7e711742cd6cbfca997cb4da3b300183cbdbc7dee350fadc6f63b40b8": plugin type="cilium-cni" name="cilium" failed (add): Unable to create endpoint: Cilium API client timeout exceeded
  Warning  FailedCreatePodSandBox  32m                 kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "513188534585a418cae73dc28d60feb86cc03caa7be3c4850c25c1419db6f923": plugin type="cilium-cni" name="cilium" failed (add): Unable to create endpoint: Cilium API client timeout exceeded
  Warning  FailedCreatePodSandBox  30m                 kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "64da6480f7e2ae422b9c09ddab7155ed7581b8b2b1363c1806786d05c74fa949": plugin type="cilium-cni" name="cilium" failed (add): Unable to create endpoint: Cilium API client timeout exceeded
  Warning  FailedCreatePodSandBox  29m                 kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "2ce285e759e1de492fe9cf59db8c61e15943ce8d6b120d7fb78afd39c09a025e": plugin type="cilium-cni" name="cilium" failed (add): Unable to create endpoint: Cilium API client timeout exceeded
  Warning  FailedCreatePodSandBox  27m                 kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "794e70456e6452e9f83755e0cc983de762e7daef741456a02086f87fbc3ca8fa": plugin type="cilium-cni" name="cilium" failed (add): Unable to create endpoint: Cilium API client timeout exceeded
  Warning  FailedCreatePodSandBox  25m                 kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "29c8a1c6c61db607ebb9ccea2fd81255d097742a20783ea897b66ae9be1a07e3": plugin type="cilium-cni" name="cilium" failed (add): Unable to create endpoint: Cilium API client timeout exceeded
  Warning  FailedCreatePodSandBox  23m                 kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "63d4be1b29016be01eee8902eaf0a54261c03b01999162d134a63aefa625e505": plugin type="cilium-cni" name="cilium" failed (add): Unable to create endpoint: Cilium API client timeout exceeded
  Warning  FailedCreatePodSandBox  22m                 kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "712fdb265f0c236bfef2a16e7d487efa53c53e5a2b5e04aed02674393674203d": plugin type="cilium-cni" name="cilium" failed (add): Unable to create endpoint: Cilium API client timeout exceeded
  Warning  FailedCreatePodSandBox  20m                 kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "c7845ce1b5447031d2d5d2af0e73f2adb80f7a515bd2253becb09afbdd62be87": plugin type="cilium-cni" name="cilium" failed (add): Unable to create endpoint: Cilium API client timeout exceeded
  Warning  FailedCreatePodSandBox  87s (x11 over 18m)  kubelet            (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "15e5967c08ad01fd11c2f343a65a6a468d9b957257d737b79b646d7f3af88a5b": plugin type="cilium-cni" name="cilium" failed (add): Unable to create endpoint: Cilium API client timeout exceeded
```
Here is cilium log
```
level=warning msg="1607: (6b) *(u16 *)(r10 -144) = r2" subsys=datapath-loader
level=warning msg="1608: (69) r2 = *(u16 *)(r10 -158)" subsys=datapath-loader
level=warning msg="BPF program is too large. Processed 1000001 insn" subsys=datapath-loader
level=warning msg="processed 1000001 insns (limit 1000000) max_states_per_insn 22 total_states 65677 peak_states 1966 mark_read 85" subsys=datapath-loader
level=warning subsys=datapath-loader
level=warning msg="JIT disabled, but 2/2 tail call maps in the program have JITed owner!" subsys=datapath-loader
level=warning msg="Error filling program arrays!" subsys=datapath-loader
level=warning msg="Unable to load program" subsys=datapath-loader
level=warning msg="JoinEP: Failed to load program" containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=2599 error="Failed to load prog with tc: exit status 1" file-path=2599_next/bpf_lxc.o identity=4 ipv4= ipv6= k8sPodName=/ subsys=datapath-loader veth=lxc_health
level=error msg="Error while rewriting endpoint BPF program" containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=2599 error="Failed to load prog with tc: exit status 1" identity=4 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=warning msg="generating BPF for endpoint failed, keeping stale directory." containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=2599 file-path=2599_next_fail identity=4 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=warning msg="Regeneration of endpoint failed" bpfCompilation=0s bpfLoadProg=15.022215765s bpfWaitForELF="37.661µs" bpfWriteELF=1.146148ms containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=2599 error="Failed to load prog with tc: exit status 1" identity=4 ipv4= ipv6= k8sPodName=/ mapSync="3.146µs" policyCalculation="3.707µs" prepareBuild="836.244µs" proxyConfiguration="8.015µs" proxyPolicyCalculation="11.942µs" proxyWaitForAck=0s reason="retrying regeneration" subsys=endpoint total=15.027303051s waitingForCTClean=2.202727ms waitingForLock=652ns
level=error msg="endpoint regeneration failed" containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=2599 error="Failed to load prog with tc: exit status 1" identity=4 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Releasing key" key="[k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name=kube-system k8s:io.cilium.k8s.policy.cluster=default k8s:io.cilium.k8s.policy.serviceaccount=coredns k8s:io.kubernetes.pod.namespace=kube-system k8s:k8s-app=kube-dns]" subsys=allocator
level=info msg="Removed endpoint" containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=109 identity=16514 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=warning msg="Creation of endpoint failed" containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=109 error="timeout while waiting for initial endpoint generation to complete" identity=16514 ipv4= ipv6= k8sPodName=/ subsys=daemon
level=info msg="API call has been processed" error="timeout while waiting for initial endpoint generation to complete" name=endpoint-create processingDuration=1m30.002332666s subsys=rate totalDuration=1m30.002410502s uuid=0f2772c2-e607-4bea-99f9-fa484f961b10 waitDurationTotal="64.341µs"
level=info msg="Processing API request with rate limiter" name=endpoint-delete parallelRequests=4 subsys=rate uuid=fd77e2ed-4322-4398-bd2f-af12cf3721b8
level=info msg="API request released by rate limiter" name=endpoint-delete parallelRequests=4 subsys=rate uuid=fd77e2ed-4322-4398-bd2f-af12cf3721b8 waitDurationTotal="44.704µs"
level=info msg="API call has been processed" error="endpoint not found" name=endpoint-delete processingDuration="2.645µs" subsys=rate totalDuration="59.042µs" uuid=fd77e2ed-4322-4398-bd2f-af12cf3721b8 waitDurationTotal="44.704µs"
level=info msg="Processing API request with rate limiter" maxWaitDuration=15s name=endpoint-create parallelRequests=4 subsys=rate uuid=529167d0-82e9-4633-8614-e722c84a085c
level=info msg="API request released by rate limiter" burst=4 limit=0.50/s maxWaitDuration=15s maxWaitDurationLimiter=14.999945187s name=endpoint-create parallelRequests=4 subsys=rate uuid=529167d0-82e9-4633-8614-e722c84a085c waitDurationLimiter=0s waitDurationTotal="61.576µs"
level=info msg="Create endpoint request" addressing="&{10.0.0.94 22002b89-cb1d-454c-9005-5a46987ad0ae  }" containerID=ce893f9261f48278eea6f5422644349dee938d61c11e1a851abbf65447ef811a datapathConfiguration="<nil>" interface=lxce5c9c9963441 k8sPodName=kube-system/coredns-8558b7dd95-t76zd labels="[]" subsys=daemon sync-build=true
level=info msg="New endpoint" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=2851 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Resolving identity labels (blocking)" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=2851 identityLabels="k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name=kube-system,k8s:io.cilium.k8s.policy.cluster=default,k8s:io.cilium.k8s.policy.serviceaccount=coredns,k8s:io.kubernetes.pod.namespace=kube-system,k8s:k8s-app=kube-dns" ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Reusing existing global key" key="k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name=kube-system;k8s:io.cilium.k8s.policy.cluster=default;k8s:io.cilium.k8s.policy.serviceaccount=coredns;k8s:io.kubernetes.pod.namespace=kube-system;k8s:k8s-app=kube-dns;" subsys=allocator
level=info msg="Identity of endpoint changed" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=2851 identity=16514 identityLabels="k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name=kube-system,k8s:io.cilium.k8s.policy.cluster=default,k8s:io.cilium.k8s.policy.serviceaccount=coredns,k8s:io.kubernetes.pod.namespace=kube-system,k8s:k8s-app=kube-dns" ipv4= ipv6= k8sPodName=/ oldIdentity="no identity" subsys=endpoint
level=info msg="Waiting for endpoint to be generated" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=2851 identity=16514 ipv4= ipv6= k8sPodName=/ subsys=endpoint
```

**Comment by mnaser** [comment]
@okozachenko1203 I think this is related to https://github.com/vexxhost/atmosphere/issues/423 -- if we update it, it'll probably work.
